### ISSUE TITLE
Completion from unimported namespaces doesn't honour implicit usings

### ISSUE THREAD

**Issue by davidwengier** [dotnet/roslyn#58473]
**Version Used**: Version 17.1.0 Preview 3.0 [32015.112.main]

**Steps to Reproduce**:

1. New C# Console app
2. Turn on "Show completion from unimported namespaces"
2. Type `Enviro`
3. Wait for completion

**Expected Behavior**:
Completion offers `Environment` because `using System;` is implicit in .NET 6 console apps

**Actual Behavior**:
Completion offers duplicate `Environment` entries, one of which adds a redundant using to the file.

![ImplicitUsingsBug](https://user-images.githubusercontent.com/754264/147291669-d0bdb930-c062-4ed0-b3e6-44d9f7fed2db.gif)

**Comment by genlu** [comment]
@AlekseyTs Do we have APIs to get all global usings from given a compilation?

**Comment by AlekseyTs** [comment]
> Do we have APIs to get all global usings from given a compilation?

I do not think so. I also think that we don't have an API like that for regular usings.

**Comment by CyrusNajmabadi** [comment]
@genlu  if necessary, we could likely easily expose this from teh decl table.

**Comment by genlu** [comment]
> if necessary, we could likely easily expose this from teh decl table.

@CyrusNajmabadi that'd be great, do we need a proposal for this?

**Comment by genlu** [comment]
@CyrusNajmabadi we should pursue adding the API for global usings, but @tmeschter just suggested we could use a heuristic that looking for SDK generated file. This as a targeted fix, would be extremely simple yet cover a lot of cases for us.

**Comment by CyrusNajmabadi** [comment]
>  but @tmeschter just suggested we could use a heuristic that looking for SDK generated file.

Could you expand on this?

**Comment by genlu** [comment]
Basically, we could just look for a document called `*.GlobalUsings.g.cs`, which is the default name generated by SDK

![image](https://user-images.githubusercontent.com/788783/148310367-588a4665-be03-49aa-8f5c-91faa296d6d6.png)

**Comment by CyrusNajmabadi** [comment]
Got it.  Thanks!

**Comment by genlu** [comment]
Move this to backlog, blocked by compiler API for global usings
